### ISSUE TITLE
Incorrect Unnecessary escape character warning

### ISSUE THREAD

**Issue by krisfields** [eslint/eslint#8592]
eslint version 3.16.1
node version 6.6.0
npm version 3.10.8

For my slugify function below, I'm getting 3 unnecessary escape character warnings on /- when those parts are definitely needed.

```js
function slugify(text) {
  return text.toString().toLowerCase()
    .replace(/\s+/g, '-')           // Replace spaces with -
    .replace(/[^\w\-]+/g, '')       // Remove all non-word chars
    .replace(/\-\-+/g, '-')         // Replace multiple - with single -
    .replace(/^-+/, '')             // Trim - from start of text
    .replace(/-+$/, '');            // Trim - from end of text
}
```

----
01:32:54 web.1  |    25:19  warning  Unnecessary escape character: \-               no-useless-escape
01:32:54 web.1  |    26:15  warning  Unnecessary escape character: \-               no-useless-escape
01:32:54 web.1  |    26:17  warning  Unnecessary escape character: \-               no-useless-escape

**Comment by ljharb** [comment]
Why are they needed? `-` doesn't need to be escaped when it couldn't possibly be part of a range.

Which specific lines are warning? (the line numbers aren't apparent)

**Comment by mysticatea** [comment]
Thank you for this issue.

But I'm confused.
Your code does not have `\-`, so I think `no-useless-escape` reports nothing.
I cannot find problems on your code.

![image](https://cloud.githubusercontent.com/assets/1937871/26030702/20a27ee4-3895-11e7-80fa-47546b9594e8.png)

**Comment by ljharb** [comment]
@mysticatea edit their OP, and surround their code in triple backticks (as they should have done) and i'll bet they'll show up.

**Comment by mysticatea** [comment]
@ljharb Ah, thank you!

So, this is working correctly. Those escapes are unnecessary by the language spec.

**Comment by ilyavolodin** [comment]
Closing, as this is correct behavior, based on the regex spec.

**Comment by krisfields** [comment]
I'm sorry.  I am confused.  How am I supposed to replace all non word characters except -?  And also, how am I supposed to replace double - with a single -?  That's what the two lines do, and if I remove the \-, they no longer do that.

before (desired behavior):
slugify("this--is-a--test") = this-is-a-test

after satisfying eslint (incorrect behavior):
slugify("this--is-a--test") = thisisatest

Thanks!

**Comment by ljharb** [comment]
(Please use backticks around code, so we can see your backslashes)

`[^w-]` is all non-word chars except the dash.

`/--/` is two backticks in a row.

If you make a jsfiddle that shows the problem, I can help you make it work.

**Comment by krisfields** [comment]
Sorry about the backticks.

Here's a jsfiddle with the slugify function before and after.
[https://jsfiddle.net/a21vjk8u/](url)

**Comment by krisfields** [comment]
Ok, sorry back the backticks.  I was confused as to how to fix.  I figured it out now.  Thank you.
